src/cargo/core/compiler/custom_build.rs

```diff
@@ -725,13 +725,26 @@ impl BuildOutput {
         fn check_minimum_supported_rust_version_for_new_syntax(
             pkg_descr: &str,
             msrv: &Option<RustVersion>,
+            key: &str,
         ) -> CargoResult<()> {
             if let Some(msrv) = msrv {
                 let new_syntax_added_in = RustVersion::from_str("1.77.0")?;
                 if !new_syntax_added_in.is_compatible_with(msrv.as_partial()) {
+                    let prefix = format!("{key}=");
+
+                    let old_syntax_suggestion = RESERVED_PREFIXES
+                        .contains(&&*prefix)
+                        .then(|| {
+                            format!(
+                                "Consider using the old `cargo:` syntax in front of `{prefix}`.\n"
```

Comment:
imo splitting up the suggestion like this feels a little more awkward to read.

Maybe

```suggestion
                                "Switch to `cargo:{prefix}` (note the single colon).\n"
```

Reply:
I wanted to make it clear that the single colon is an old syntax. What do you think?

Reply:
OK, how about now?